lnwallet/channel.go

```diff
@@ -1945,52 +1945,61 @@ func (lc *LightningChannel) ForceClose() (*ForceCloseSummary, error) {
 		return nil, err
 	}
 
+	// Re-derive the original pkScript for to-self output within the
+	// commitment transaction. We'll need this to find the corresponding
+	// output in the commitment transaction and potentially for creating
+	// the sign descriptor.
+	csvTimeout := lc.channelState.LocalCsvDelay
+	selfKey := lc.channelState.OurCommitKey
+	elkrem := lc.channelState.LocalElkrem
+	unusedRevocation, err := elkrem.AtIndex(lc.currentHeight)
+	if err != nil {
+		return nil, err
+	}
+	revokeKey := DeriveRevocationPubkey(lc.channelState.TheirCommitKey,
+		unusedRevocation[:])
+	selfScript, err := commitScriptToSelf(csvTimeout, selfKey, revokeKey)
+	if err != nil {
+		return nil, err
+	}
+	payToUsScriptHash, err := witnessScriptHash(selfScript)
+	if err != nil {
+		return nil, err
+	}
+
 	// Locate the output index of the delayed commitment output back to us.
 	// We'll return the details of this output to the caller so they can
 	// sweep it once it's mature.
 	// TODO(roasbeef): also return HTLC info, assumes only p2wsh is commit
 	// tx
 	var delayIndex uint32
 	var delayScript []byte
+	var selfSignDesc *SignDescriptor
 	for i, txOut := range commitTx.TxOut {
-		if !txscript.IsPayToWitnessScriptHash(txOut.PkScript) {
+		if !bytes.Equal(payToUsScriptHash, txOut.PkScript) {
 			continue
 		}
 
 		delayIndex = uint32(i)
 		delayScript = txOut.PkScript
-	}
-
-	csvTimeout := lc.channelState.LocalCsvDelay
-	selfKey := lc.channelState.OurCommitKey
-
-	// Re-derive the original pkScript for out to-self output within the
-	// commitment transaction. We'll need this for the created sign
-	// descriptor.
-	elkrem := lc.channelState.LocalElkrem
-	unusedRevocation, err := elkrem.AtIndex(lc.currentHeight)
-	if err != nil {
-		return nil, err
-	}
-	revokeKey := DeriveRevocationPubkey(lc.channelState.TheirCommitKey,
-		unusedRevocation[:])
-	selfScript, err := commitScriptToSelf(csvTimeout, selfKey, revokeKey)
-	if err != nil {
-		return nil, err
+		break
```

Comment:
Nice! This also fixes an existing bug atm which can arise if a channel is forced closed with uncleared HTLC's